code/game/machinery/teleporter/console.dm

```diff
@@ -51,66 +48,57 @@
 			projector.queue_icon_update()
 			pad.queue_icon_update()
 
-
 /obj/machinery/computer/teleporter/proc/clear_projector()
 	if (!projector)
 		return
-	GLOB.destroyed_event.unregister(projector, src, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_projector))
+	UnregisterSignal(projector, COMSIG_PARENT_QDELETING)
 	projector = null
 	set_active(FALSE)
 
-
 /obj/machinery/computer/teleporter/proc/lost_projector()
 	audible_message(SPAN_WARNING("\The [src] buzzes, \"Projector missing.\""))
 	clear_projector()
 
-
 /obj/machinery/computer/teleporter/proc/set_projector(obj/machinery/tele_projector/_projector)
 	if (projector == _projector)
 		return
 	clear_projector()
 	projector = _projector
-	GLOB.destroyed_event.register(projector, src, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_projector))
-
+	RegisterSignal(projector, COMSIG_PARENT_QDELETING, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_projector))
 
 /obj/machinery/computer/teleporter/proc/clear_pad()
 	if (!pad)
 		return
-	GLOB.destroyed_event.unregister(pad, src, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_pad))
+	UnregisterSignal(pad, COMSIG_PARENT_QDELETING)
 	pad = null
 	set_active(FALSE)
 
-
 /obj/machinery/computer/teleporter/proc/lost_pad()
 	audible_message(SPAN_WARNING("\The [src] buzzes, \"Pad missing.\""))
 	clear_pad()
 
-
 /obj/machinery/computer/teleporter/proc/set_pad(obj/machinery/tele_pad/_pad)
 	if (pad == _pad)
 		return
 	clear_pad()
 	pad = _pad
-	GLOB.destroyed_event.register(pad, src, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_pad))
-
+	RegisterSignal(pad, COMSIG_PARENT_QDELETING, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_pad))
 
 /obj/machinery/computer/teleporter/proc/clear_target()
 	if (!target)
 		return
 	var/old_target = target
-	GLOB.destroyed_event.unregister(target, src, TYPE_PROC_REF(/obj/machinery/computer/teleporter, lost_target))
+	UnregisterSignal(target, COMSIG_PARENT_QDELETING)
```

Comment:
you do need to know that this is a thing and I'm not sure if unregistering ALL signals instead of one specific proc would break anything, but I'd prefer you to keep the things where it specified proc to unregister signal from

Reply:
If you dont want to do it, then convince me in discord and I'll approve it